src/Microsoft.Framework.DependencyInjection.Abstractions/ServiceDescriptor.cs

```diff
@@ -22,6 +23,14 @@ public class ServiceDescriptor
                                  ServiceLifetime lifetime)
             : this(serviceType, lifetime)
         {
+            if (serviceType.GetTypeInfo().IsGenericTypeDefinition &&
```

Comment:
Thoughts on verifying it in the `ServiceDescriptor`? Seemed like the one place to all the registrations get funneled to.